dowhy/causal_refuters/overrule/BCS/beam_search.py

```diff
@@ -0,0 +1,201 @@
+"""Beam search utilities for optimization.
+
+This module implements the boolean ruleset estimator from OverRule [1]. Code is adapted (with some simplifications)
+from https://github.com/clinicalml/overlap-code, under the MIT License.
+
+[1] Oberst, M., Johansson, F., Wei, D., Gao, T., Brat, G., Sontag, D., & Varshney, K. (2020). Characterization of
+Overlap in Observational Studies. In S. Chiappa & R. Calandra (Eds.), Proceedings of the Twenty Third International
+Conference on Artificial Intelligence and Statistics (Vol. 108, pp. 788–798). PMLR. https://arxiv.org/abs/1907.04138
+"""
+
+import numpy as np
+import pandas as pd
+
+
+class PricingInstance:
+    """
+    Instance of the pricing problem.
```

Comment:
What is the pricing problem?  I wonder if there's a name for this class that would be more intuitive in a causal inference context.  Alternatively, it would be sufficient to point to a formal description of the pricing problem

Reply:
Added a citation - this doesn't mean much in a causal inference context, it's an implementation detail of how the optimization problem is solved.